### PR TITLE
AP_MATH: Remove isnan check for non-float type

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

File: libraries/AP_Math/AP_Math.cpp [hunk 1]
```diff
@@ -265,9 +265,11 @@ T constrain_value(const T amt, const T low, const T high)
     // the check for NaN as a float prevents propagation of floating point
     // errors through any function that uses constrain_value(). The normal
     // float semantics already handle -Inf and +Inf
-    if (isnan(amt)) {
-        INTERNAL_ERROR(AP_InternalError::error_t::constraining_nan);
-        return (low + high) / 2;
+    if (!std::is_integral<T>::value) {
```
Comment (peterbarker) [inline libraries/AP_Math/AP_Math.cpp:268]: Why not `is_floating_point`?

**Review by peterbarker — CHANGES_REQUESTED** [review]
Commit message needs fixing - `AP_Math: only check floating point numbers for isnan`

File: libraries/AP_Math/AP_Math.cpp (hunk 1 above)
Comment (patrickelectric) [reply libraries/AP_Math/AP_Math.cpp]: done.

**Comment by peterbarker** [comment]
Tweaked the commit message (AP_Math vs AP_MATH) and merged.

Thanks!

**Pull request merged**